sonic-slave-buster/Dockerfile.j2

```diff
@@ -40,11 +48,86 @@ RUN echo "deb [arch=arm64] http://deb.debian.org/debian buster main contrib non-
         echo 'deb [arch=arm64] http://ftp.debian.org/debian buster-backports main' >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian buster main contrib non-free" >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian buster-updates main contrib non-free" >> /etc/apt/sources.list
+{%- elif CONFIGURED_ARCH == "armhf" and CROSS_BUILD_ENVIRON == "y" %}
+RUN dpkg --add-architecture armhf
+RUN apt-get update 
+RUN apt-get install -y crossbuild-essential-armhf
+RUN apt-get install -y gcc-arm-linux-gnueabihf
+ENV $(dpkg-architecture -a$(CONFIGURED_ARCH))
+
+RUN apt-mark hold crossbuild-essential-armhf
+RUN apt-mark hold g++-arm-linux-gnueabihf
+RUN apt-mark hold g++-8-arm-linux-gnueabihf
+RUN apt-mark hold gcc-arm-linux-gnueabihf
+RUN apt-mark hold gcc-8-arm-linux-gnueabihf
+
+ARG arch=armhf
+ARG CROSS_CC=arm-linux-gnueabihf-gcc
+ARG CROSS_CXX=arm-linux-gnueabihf-g++
+{%- elif CONFIGURED_ARCH == "arm64" and CROSS_BUILD_ENVIRON == "y" %}
+RUN dpkg --add-architecture arm64
+RUN apt-get update
+RUN apt-get install -y crossbuild-essential-arm64
+RUN apt-get install -y gcc-aarch64-linux-gnu
+ENV $(dpkg-architecture -a$(CONFIGURED_ARCH))
+
+RUN apt-mark hold crossbuild-essential-arm64
+RUN apt-mark hold g++-aarch64-linux-gnu
+RUN apt-mark hold g++-8-aarch64-linux-gnu
+RUN apt-mark hold gcc-aarch64-linux-gnu
+RUN apt-mark hold gcc-8-aarch64-linux-gnu
+ARG arch=arm64
+ARG CROSS_CC=aarch64-linux-gnu-gcc
+ARG CROSS_CXX=aarch64-linux-gnu-g++
 {%- endif %}
 
 ## Make apt-get non-interactive
 ENV DEBIAN_FRONTEND=noninteractive
 
+{%- if CROSS_BUILD_ENVIRON == "y" %}
+RUN apt-get update && apt-get install -y python3 python3-pip
+
+RUN apt-get update && apt-get install -y \
+	python:$arch python3-minimal:$arch python3.7:$arch python3:$arch python-setuptools:$arch 
+RUN apt-get download python3-distutils && dpkg --force-all -i python3-distutils*
+RUN apt-get download python3-pip && dpkg --force-all -i python3-pip*
+RUN which pip3 && pip3 install enum34
+RUN pip3 install virtualenv
+
+RUN file /usr/bin/python2.7 && file /usr/bin/python3.7
+
+# Create target arm python2/3 virtual environments with all required packages installed
+RUN mkdir /python_virtualenv
+RUN cd /python_virtualenv && python3 -m virtualenv -p /usr/bin/python env2
+RUN cd /python_virtualenv && python3 -m virtualenv --copies -p /usr/bin/python3 env3
+
+RUN PATH=/python_virtualenv/env2/bin/:$PATH pip2 install setuptools==40.8.0 wheel==0.35.1 fastentrypoints pytest pytest-cov pytest-runner==4.4 nose==1.3.7 mockredispy==2.9.3 mock==3.0.5 j2cli==0.3.10 PyYAML==5.4.1 pexpect==4.6.0 Pympler==0.8 ctypesgen==1.0.2 natsort redis
```

Comment:
What happens when the pip package versions obsolete ?

Reply:
> What happens when the pip package versions obsolete ?

These are python dependencies specific versions for building Sonic packages. They did exist in the Dockerfile before my change. I guess that they have to be updated if one of the Sonic packages requires it.